datacube.py

```diff
@@ -0,0 +1,436 @@
+
+    # _, index = np.unique(da_dem['time'], return_index=True)  # Remove redundant time
+    # da_dem = da_dem.isel(time=index)
+
+    return da_sen2_all, ds_sen1, da_dem
+
+
+def merge_datarrays(da_sen2, da_sen1, da_dem):
+    """
+    Merge xarray DataArrays for Sentinel-2, Sentinel-1, and DEM.
+
+    Parameters:
+    - da_sen2 (xr.DataArray): xarray DataArray for Sentinel-2 data.
+    - da_sen1 (xr.DataArray): xarray DataArray for Sentinel-1 data.
+    - da_dem (xr.DataArray): xarray DataArray for DEM data.
+
+    Returns:
+    - xr.DataArray: Merged xarray DataArray.
+    """
+    # print("Platform variables (S2, S1, DEM): ", da_sen2.platform.values, da_sen1.platform.values, da_dem.platform.values)
+    # da_sen2 = da_sen2.drop(["platform", "constellation"])
+    # da_sen1 = da_sen1.drop(["platform", "constellation"])
+    # da_dem = da_dem.drop(["platform"])
+
+    da_merge = xr.merge([da_sen2, da_sen1, da_dem], compat="override")
+    print("Merged datarray: ", da_merge)
+    print(
+        "Time variables (S2, merged): ", da_sen2.time.values, da_merge.time.values
+    )  # da_sen1.time.values, da_dem.time.values
+    return da_merge
+
+
+def process(
+    year1, year2, aoi, resolution, cloud_cover_percentage, nodata_pixel_percentage
+):
+    """
+    Process Sentinel-2, Sentinel-1, and DEM data for a specified time range, area of interest (AOI),
+    resolution, EPSG code, cloud cover percentage, and nodata pixel percentage.
+
+    Parameters:
+    - year1 (int): The starting year of the date range.
+    - year2 (int): The ending year of the date range.
+    - aoi (shapely.geometry.base.BaseGeometry): Geometry object for an Area of Interest (AOI).
+    - resolution (int): Spatial resolution.
+    - epsg (int): EPSG code for the coordinate reference system.
+    - cloud_cover_percentage (int): Maximum acceptable cloud cover percentage for Sentinel-2 images.
+    - nodata_pixel_percentage (int): Maximum acceptable percentage of nodata pixels in Sentinel-2 images.
+
+    Returns:
+    - xr.DataArray: Merged xarray DataArray containing processed data.
+    """
+
+    date, YEAR, MONTH, DAY, CLOUD = get_conditions(year1, year2)
+    week = get_week(YEAR, MONTH, DAY)
+
+    catalog, s2_items, BBOX, epsg = search_sentinel2(
+        week, aoi, cloud_cover_percentage, nodata_pixel_percentage
+    )
+
+    s1_items = search_sentinel1(BBOX, catalog, week)
+
+    dem_items = search_dem(BBOX, catalog, epsg)
+
+    da_sen2, da_sen1, da_dem = make_dataarrays(
+        s2_items, s1_items, dem_items, BBOX, resolution, epsg
+    )
+
+    da_merge = merge_datarrays(da_sen2, da_sen1, da_dem)
+    return da_merge
+
+
+# EXAMPLE
+california_tile = gpd.read_file("ca.geojson")
+sample = california_tile.sample(1)
+aoi = sample.iloc[0].geometry
+cloud_cover_percentage = 50
+nodata_pixel_percentage = 20
+merged = process(
+    2017, 2023, aoi, 10, cloud_cover_percentage, nodata_pixel_percentage
+)  # Spatial resolution of 10 metres
```

Comment:
Wrap this block in `if __name__ == "__main__":` so that we can import functions from this datacube.py file without triggerring the datacube processing.
```suggestion
if __name__ == "__main__":
    california_tile = gpd.read_file("ca.geojson")
    sample = california_tile.sample(1)
    aoi = sample.iloc[0].geometry
    cloud_cover_percentage = 50
    nodata_pixel_percentage = 20
    merged = process(
        2017, 2023, aoi, 10, cloud_cover_percentage, nodata_pixel_percentage
    )  # Spatial resolution of 10 metres
```

Reply:
Done